From d8b8ac9bf6393e7cd5f689804668eb2864e0275a Mon Sep 17 00:00:00 2001
From: Naveen Mamindlapalli <naveenm@marvell.com>
Date: Mon, 17 May 2021 13:00:14 +0530
Subject: [PATCH 7/7] octeontx2-bphy-netdev: cpri: fix compilation warnings
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit 1cff5ee90be2221994b6f9f47dbcedfcd01fd020 from
git@git.assembla.com:cavium/WindRiver.linux.git

Fixed following compilation warnings.

drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c:60:20:
warning: no previous prototype for ‘otx2_cpri_get_netdev’
	[-Wmissing-prototypes]

drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c:129:6:
warning: variable ‘sw_rd_ptr’ set but not used
	[-Wunused-but-set-variable]

Change-Id: I452b38bfd804862298db9ac06820ec59b34715da
Signed-off-by: Naveen Mamindlapalli <naveenm@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/51947
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
index 816fe2dbb3d4..2035031cb3d2 100644
--- a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
+++ b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_cpri.c
@@ -57,7 +57,7 @@
 /* global driver ctx */
 struct otx2_cpri_drv_ctx cpri_drv_ctx[OTX2_BPHY_CPRI_MAX_INTF];
 
-struct net_device *otx2_cpri_get_netdev(int mhab_id, int lmac_id)
+static struct net_device *otx2_cpri_get_netdev(int mhab_id, int lmac_id)
 {
 	struct net_device *netdev = NULL;
 	int idx;
@@ -126,16 +126,13 @@ static int otx2_cpri_process_rx_pkts(struct otx2_cpri_ndev_priv *priv,
 	struct otx2_cpri_ndev_priv *priv2;
 	struct cpri_pkt_ul_wqe_hdr *wqe;
 	struct ul_cbuf_cfg *ul_cfg;
-	u16 sw_rd_ptr, nxt_wr_ptr;
 	struct net_device *netdev;
+	u16 nxt_wr_ptr, len;
 	struct sk_buff *skb;
 	u8 *pkt_buf;
-	u16 len;
 
 	ul_cfg = &priv->cpri_common->ul_cfg;
 
-	sw_rd_ptr = readq(priv->cpri_reg_base +
-			  CPRIX_RXD_GMII_UL_SW_RD_PTR(priv->cpri_num)) & 0xFFFF;
 	nxt_wr_ptr = readq(priv->cpri_reg_base +
 			   CPRIX_RXD_GMII_UL_NXT_WR_PTR(priv->cpri_num)) &
 			0xFFFF;
-- 
2.31.1

